crates/uv-resolver/src/requires_python/tests.rs

```diff
@@ -156,3 +156,25 @@ fn is_exact_without_patch() {
         assert_eq!(requires_python.is_exact_without_patch(), expected);
     }
 }
+
+#[test]
+fn is_tilde_exact_without_patch() {
+    let test_cases = [
+        ("~=3.11", true),
+        ("~=2.7", true),
+        ("~=3.10, <3.11", false),
+        ("~=3.10, <=3.11", false),
```

Comment:
While unlikely, this sort of version should not throw a warning, correct?

Reply:
cc @konstin

Reply:
(I think not)

Reply:
We can leave it without a warning. It's an odd version specifier to write and likely not what the user intended, but it's not one that specifically matches this case.